src/v/net/server.cc

```diff
@@ -194,12 +196,22 @@ ss::future<> server::accept(listener& s) {
                   }
               }
 
+              std::optional<security::tls::principal_mapper> tls_pm;
+              auto se_it = std::find_if(
+                cfg.addrs.begin(), cfg.addrs.end(), [&name](const auto& a) {
+                    return a.name == name;
```

Comment:
is there any sort of edge case here if an address and the listener both had empty strings for name?

Reply:
I think it's ok. The names have to be unique, and empty is allowed, so the only way to match it would be with an empty name.

Also see: https://github.com/BenPope/redpanda/blob/0b98ddf4ac8a975fd04ba26339304605f3abf2cb/src/v/redpanda/application.cc#L1002-L1007

Reply:
I guess I meant like: if listener name and config name were both empty / omitted from config, would that be ambiguous and the default behavior would be expected to not have the two associated by name?

Reply:
ahh but I see that in the link you left this is basically preserving the existing behavior

Reply:
Something like this?
```yaml
  kafka_api:
    - address: 0.0.0.0
      name: local
      port: 19092
    - address: 0.0.0.0
      port: 19093

  advertised_kafka_api:
    - address: localhost
      name: local
      port: 19092
    - address: localhost
      port: 19093

  kafka_api_tls:
    - enabled: false
      name: local
    - cert_file: mtls_server.crt
      enabled: true
      key_file: mtls_server.key
      require_client_auth: true
      principal_mapping_rules: "RULE:.*CAN=([^,]+).*/$1/"
      truststore_file: mtls_ca.crt
```
Where we're interested in the not `local` listener? I think it works as expected.